crates/server/src/store/forwarder/authority.rs

```diff
@@ -88,18 +92,71 @@ impl<P: ConnectionProvider> ForwardAuthority<P> {
 
         let config = ResolverConfig::from_parts(None, vec![], name_servers);
 
-        let resolver = Resolver::new(config, options, runtime);
+        #[cfg_attr(not(feature = "__dnssec"), allow(unused_mut))]
+        let mut resolver_builder = Resolver::builder(config, options, runtime);
+
+        #[cfg(feature = "__dnssec")]
+        if let Some(trust_anchor) = trust_anchor {
+            resolver_builder = resolver_builder.with_trust_anchor(trust_anchor);
+        }
 
-        info!("forward resolver configured: {}: ", origin);
+        let resolver = resolver_builder.build();
 
-        // TODO: this might be infallible?
-        Ok(Self {
+        info!(%origin, "forward resolver configured");
+
+        Ok(ForwardAuthority {
             origin: origin.into(),
             resolver,
         })
     }
 }
 
+/// An authority that will forward resolutions to upstream resolvers.
+///
+/// This uses the hickory-resolver crate for resolving requests.
+pub struct ForwardAuthority<P: ConnectionProvider = TokioConnectionProvider> {
+    origin: LowerName,
+    resolver: Resolver<P>,
+}
+
+impl<P: ConnectionProvider> ForwardAuthority<P> {
```

Comment:
Here, too, provide a `builder(runtime: P)` method that covers the `from_system_conf()` case, and a `builder_with_config()` method for the other case. `origin` can default to `Name::root()` and be set explicitly with `with_origin()` on the builder. Looks like the `ZoneType` argument can just be dropped?